src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, maxerr: 50 */
+/*global define, $, brackets, console */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var AppInit = require("utils/AppInit"),
+        PreferencesManager = require("preferences/PreferencesManager"),
+        HealthDataUtils = require("healthData/HealthDataUtils"),
+        uuid = require("thirdparty/uuid"),
+        UrlParams = require("utils/UrlParams").UrlParams;
+
+    var prefs = PreferencesManager.getExtensionPrefs("healthData");
+
+    prefs.definePreference("healthDataTracking", "boolean", true);
+
+    var ONE_DAY = 24 * 60 * 60 * 1000,
+        timeoutVar;
+
+    var params = new UrlParams();
+    params.parse();
+
+    /**
+     * Get the health data which will be send to the server. Initially it is only one time data.
+     */
+    function getHealthData() {
+        var result = new $.Deferred(),
+            oneTimeHealthData = {};
+
+        var guid = PreferencesManager.getViewState("GUID");
+
+        if (!guid) {
+            guid = uuid.v4();
+            PreferencesManager.setViewState("GUID", guid);
+        }
+
+        oneTimeHealthData.guid = guid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.osVersion = HealthDataUtils.getOSVersion();
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (jsonData) {
+            var url;
+            if (params.get("testEnvironment")) {
+                url = brackets.config.healthDataTestServerURL;
+            } else {
+                url = brackets.config.healthDataServerURL;
+            }
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: jsonData,
```

Comment:
I don't think this is going to serialize the JSON correctly -- I think you ned to explicitly call `JSON.stringify()`, or else it will be converted into a set of query parameters instead.